runtime/caml/alloc.h

```diff
@@ -86,6 +87,10 @@ Caml_inline value caml_field_boxed (value arg) { return Field (arg, 0); }
 #define caml_alloc_unboxable caml_alloc_boxed
 #define caml_field_unboxable caml_field_boxed
 
+/* Special constructor for atomics that should have exclusive ownership of
+  cache line. */
+CAMLextern value caml_atomic_make_aligned(value);
+
```

Comment:
In general, primitives (functions defined `CAMLprim`) are not declared in header files, because they are called from OCaml code only.

Here, we have a function that could reasonably be called from C code, so it's less clear.  But note for instance that there is no `caml_atomic_make` (unaligned) function that can be called from C, the closest equivalent being `caml_alloc_1(0, v)`.  

Other opinions welcome.

Reply:
Good point. We have the following functions to manipulate atomics from C, but not allocation. 

```c
memory.c:CAMLprim value caml_atomic_load (value ref)
memory.c:CAMLprim value caml_atomic_exchange (value ref, value v) /* stores are implemented as exchanges */ 
memory.c:CAMLprim value caml_atomic_cas (value ref, value oldv, value newv)
memory.c:CAMLprim value caml_atomic_fetch_add (value ref, value incr)
```

It may be useful to provide access to OCaml atomics from C, including allocations. This may be done in a separate PR. For this PR, let's remove this declaration from the header file.

Reply:
This is now removed.

Reply:
In addition I vaguely remember that there is a subtlety memory-model-wise whereby the C version of one of those should be different from the CAMLprim version.